src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -628,20 +619,32 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
         return studentList;
     }
 
-    private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
-        List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
-        return info;
+    private String getInvalidityInfoInDuplicatedEmail(String[] linesArray, int isDuplicate) {
+        if (isDuplicate > -1) {
+            return "Same email address as the student in line \"" + linesArray[isDuplicate + 1] + "\"";
+        }
+        return "";
     }
 
-    private boolean isStudentEmailDuplicated(String email,
-            List<String> studentEmailList) {
-        return studentEmailList.contains(email);
+    private void addInvalidStudentInfo(List<String> invalidityInfo, StudentAttributes student, String sanitizedLine) {
+        String info = StringHelper.toString(SanitizationHelper.sanitizeForHtml(student.getInvalidityInfo()),
+                "<br>" + Const.StatusMessages.ENROLL_LINES_PROBLEM_DETAIL_PREFIX + " ");
+        invalidityInfo.add(String.format(Const.StatusMessages.ENROLL_LINES_PROBLEM, sanitizedLine, info));
     }
 
-    private boolean isInEnrollList(StudentAttributes student,
-            List<StudentAttributes> studentInfoList) {
+    /**
+     * Check for the duplicate email, if duplicate is found returns its index in studentList else return -1.
```

Comment:
We need better header comments for this :P Just use mine:
`Returns the index of the first occurrence of the duplicate {@code email} in {@code studentList}, or -1 if {@code email} is not a duplicate in {@code studentList}.`

Copied from ArrayList#indexOf(Object) :P

@whipermr5 you ok with this format?